spec.bs

```diff
@@ -427,3 +706,49 @@ With the above infrastructure in place, we can actually fire and handle the {{Ap
     1. Let |continue| be the result of [=firing a navigate event=] at |appHistory| given <var ignore>url</var>, with <i>[=fire a navigate event/isPush=]</i> set to false, <i>[=fire a navigate event/isSameDocument=]</i> set to |isSameDocument|, <i>[=fire a navigate event/userInvolvement=]</i> set to |userInvolvement|, and <i>[=fire a navigate event/isHistoryTraversal=]</i> set to true.
     1. If |continue| is false, abort these steps.
 </div>
+
+<h2 id="session-history-patches">Patches to session history</h2>
+
+This section details monkeyaptches to [[!HTML]] to track appropriate data for associating an {{AppHistory}} with a [=session history entry=].
+
+<h3 id="session-history-new-she-fields">New [=session history entry=] items</h3>
+
+Each [=session history entry=] gains the following new [=struct/items=]:
+
+* <dfn for="session history entry">origin</dfn>, an [=origin=]
+
+* <dfn for="session history entry">app history key</dfn>, a string, initially set to the result of [=generating a random UUID=]
+
+* <dfn for="session history entry">app history id</dfn>, a string, initially set to the result of [=generating a random UUID=]
+
+* <dfn for="session history entry">app history state</dfn>, which is [=serialized state=] or null, initially null
+
+<h3 id="session-history-patches-key">Carrying over the app history key</h3>
+
+Update the <a spec="HTML">update the session history with the new page</a> algorithm's "<a for="history handling behavior">`replace`</a>" case to set <var ignore>newEntry</var>'s [=session history entry/app history key=] to <var ignore>sessionHistory</var>'s [=session history/current entry=]'s [=session history entry/app history key=].
+
+<h3 id="session-history-patches-origin">Tracking the [=session history entry/origin=] member</h3>
+
+Update the <a spec="HTML">update the session history with the new page</a> algorithm to set the [=session history entry=]'s [=session history entry/origin=] to <var ignore>newDocument</var>'s [=Document/origin=].
```

Comment:
This should mention that we only copy the key if origins match?

Reply:
Great catch; done, thanks.